### PR TITLE
Refactor Dockerfile and scripts

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Refactor Dockerfile and scripts.

**Why?**
1. Run ES index template creator only for auto setup.
2. Invoke scripts in more natural order: `entrypoint.sh` -> `auto-setup.sh` (optional), `setup-develop.sh` (optional), `start-temporal.sh` -> `exec temporal-server`.
3. Copy only needed files for every image.
4. Copy scripts to `/etc/temporal` instead of the root.
5. Some renames.
6. Comply with `shellcheck`.

**How did you test it?**
Build locally and run with docker-compose.

**Potential risks**
`RF` env was renamed to `CASSANDRA_REPLICATION_FACTOR`.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: docker/auto-setup.sh [hunk 1]
```diff
@@ -1,37 +1,34 @@
...
 
 DB="${DB:-cassandra}"
+
+# Cassandra
+KEYSPACE="${KEYSPACE:-temporal}"
+VISIBILITY_KEYSPACE="${VISIBILITY_KEYSPACE:-temporal_visibility}"
+CASSANDRA_PORT="${CASSANDRA_PORT:-9042}"
+CASSANDRA_REPLICATION_FACTOR=${CASSANDRA_REPLICATION_FACTOR:-1}
+
+# MySQL
```
Comment (wxing1292) [inline docker/auto-setup.sh:13]: # MySQL / PostgreSQL

**Review by wxing1292 — APPROVED** [review]

File: docker/auto-setup.sh [hunk 2]
```diff
@@ -1,37 +1,113 @@
...
 
-set -x
+set -ex
+
+# === Auto setup defaults ===
 
 DB="${DB:-cassandra}"
+
+# Cassandra
+KEYSPACE="${KEYSPACE:-temporal}"
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply docker/auto-setup.sh:10]: no.

File: docker/auto-setup.sh [hunk 3]
```diff
@@ -1,37 +1,113 @@
...
+        fi
+    else
+        echo "Unsupported DB type: $DB."
+        exit 1
+    fi
+}
+
+wait_for_cassandra() {
+    # TODO (alex): Remove exports
+    export CASSANDRA_USER=$CASSANDRA_USER
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply docker/auto-setup.sh:58]: I would leave it as is for now and refactor later as part of `TODO` above.

File: docker/auto-setup.sh [hunk 4]
```diff
@@ -1,37 +1,113 @@
...
+    export CASSANDRA_PORT=$CASSANDRA_PORT
+    export CASSANDRA_ENABLE_TLS=$CASSANDRA_TLS_ENABLED
+    export CASSANDRA_TLS_CERT=$CASSANDRA_CERT
+    export CASSANDRA_TLS_KEY=$CASSANDRA_CERT_KEY
+    export CASSANDRA_TLS_CA=$CASSANDRA_CA
+
+    { export CASSANDRA_PASSWORD=$CASSANDRA_PASSWORD; } 2> /dev/null
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply docker/auto-setup.sh:65]: also, in another PR.

**Review by underrun — APPROVED** [review]

**Pull request merged**